db/keyval/etcd/plugin_impl_etcd.go

```diff
@@ -154,6 +183,43 @@ func (plugin *Plugin) Compact(rev ...int64) (toRev int64, err error) {
 	return 0, fmt.Errorf("connection is not established")
 }
 
+// If ETCD is connected, complete all other procedures
+func (plugin *Plugin) configureConnection(etcdCfg *Config) {
+	plugin.reconnectResync = etcdCfg.ReconnectResync
+	if etcdCfg.AutoCompact > 0 {
+		if etcdCfg.AutoCompact < time.Duration(time.Minute*60) {
+			plugin.Log.Warnf("Auto compact option for ETCD is set to less than 60 minutes!")
+		}
+		plugin.startPeriodicAutoCompact(etcdCfg.AutoCompact)
+	}
+	plugin.protoWrapper = kvproto.NewProtoWrapperWithSerializer(plugin.connection, &keyval.SerializerJSON{})
+
+	// Register for providing status reports (polling mode).
+	if plugin.StatusCheck != nil {
+		plugin.StatusCheck.Register(core.PluginName(plugin.PluginName), func() (statuscheck.PluginState, error) {
```

Comment:
This should definitely be registered even before the connection retries, so anyone reading the status can see *etcd* there with error. You can just check if `plugin.connection` is `nil` before actually trying to call `GetValue(...)`.